Updated ScheduleWorkout Code

TharukaN99 opened the pull request:
Description :
This commit adds a new backend feature for workout scheduling, while also merging improvements and UI logic developed during the previous trimester.

New in This Commit (Current Trimester):
•	Added “Schedule” model to support storing scheduled workout plans.
•	Created “ScheduleSerializer”  for backend-to-frontend serialization.
•	Implemented two API endpoints in “views.py”:
•	“create_schedule”  – to allow workout creation.
•	“get_schedules” – to fetch a user's scheduled workouts.
•	Registered the above endpoints in “urls.py”.

Included From Previous Trimester:
The following files and logic were part of my earlier development work, carried over into this PR to consolidate functionality:
•	“edit_profile.dart” (Flutter): Notification UI adjustments and profile calendar integration.
•	Notification popup logic: Added to improve user feedback and engagement around scheduled events.
•	“main.dart” : Minor refactoring to integrate notifications and calendar view.

Design Intention:
This feature helps users plan workouts in advance and receive reminders. It’s meant to increase long-term engagement and support structured training routines.

Files Changed (Current Trimester) :
•	“models.py”: New “Schedule” model.
•	“serializers.py” : New “ScheduleSerializer”.
•	“views.py” : New endpoints “create_schedule” and “get_schedules”.
•	“urls.py” : Added routes for schedule APIs.


Files Changed (Previous Trimester) :
“edit_profile.dart” : Notification/calendar-related UI from previous term.
“main.dart” : UI entry point updated from previous term.

TharukaN99 committed:
Notification popup & Calendar

Added the Notification Popup & the Calendar to date of birth selection in edit profile

TharukaN99 committed:
Updated Notification popup

I separated the notification function from the main.dart file and created a new "nofication.dart"file under the utilities folder.

saminadhikari committed:
Merge branch 'main' into main

saminadhikari committed:
Update main.dart

TharukaN99 committed:
ScheduleWorkout

Completed Schedule workout code

TharukaN99 commented:
only approved the PR from 8th May 2025

djasmit commented:
I'd like to see some more clarification on exactly what it is this is attempting to accomplish. What files have been changed and why. The associated Planner task for this is also rather vague and lacking in comments. We need more documentation, please.

TharukaN99 commented:
Description :
This commit adds a new backend feature for workout scheduling, while also merging improvements and UI logic developed during the previous trimester.

New in This Commit (Current Trimester):
•	Added “Schedule” model to support storing scheduled workout plans.
•	Created “ScheduleSerializer”  for backend-to-frontend serialization.
•	Implemented two API endpoints in “views.py”:
•	“create_schedule”  – to allow workout creation.
•	“get_schedules” – to fetch a user's scheduled workouts.
•	Registered the above endpoints in “urls.py”.

Included From Previous Trimester:
The following files and logic were part of my earlier development work, carried over into this PR to consolidate functionality:
•	“edit_profile.dart” (Flutter): Notification UI adjustments and profile calendar integration.
•	Notification popup logic: Added to improve user feedback and engagement around scheduled events.
•	“main.dart” : Minor refactoring to integrate notifications and calendar view.

Design Intention:
This feature helps users plan workouts in advance and receive reminders. It’s meant to increase long-term engagement and support structured training routines.

Files Changed (Current Trimester) :
•	“models.py”: New “Schedule” model.
•	“serializers.py” : New “ScheduleSerializer”.
•	“views.py” : New endpoints “create_schedule” and “get_schedules”.
•	“urls.py” : Added routes for schedule APIs.


Files Changed (Previous Trimester) :
“edit_profile.dart” : Notification/calendar-related UI from previous term.
“main.dart” : UI entry point updated from previous term.

Thumbness reviewed (changes requested):
Please sync up your fork with the remote main and commit again. YOu have to resolve some conflicts

TharukaN99 committed:
FIxed git conflicts

TharukaN99 committed:
Pushing original code before fixing conflicts

TharukaN99 committed:
conflicts solutions

TharukaN99 committed:
conflicts solutions

TharukaN99 committed:
Update serializers.py

TharukaN99 committed:
Update serializers.py

TharukaN99 committed:
Update serializers.py

TharukaN99 committed:
Update serializers.py

TharukaN99 committed:
Update views.py

TharukaN99 committed:
Delete edit_profile.dart

djasmit reviewed (changes requested):
Still a couple of conflicts to sort out.

![image](https://github.com/user-attachments/assets/8ecfea16-cc85-4e25-8e7a-75b036747a61)

Where is edit_profile.dart? Why has that been removed? Was that an accident? That was an important flutter page...

The views.py conflict is trivial. Not entirely obvious what it is but easily resolved.
![image](https://github.com/user-attachments/assets/2b097fa6-0262-4428-9b73-dbde657d0324)

TharukaN99 committed:
Update views.py

TharukaN99 committed:
Revert "Delete edit_profile.dart"

This reverts commit 8b7d3f76fbd9ade38d39833ffd8765eb9b21ee1d.

TharukaN99 committed:
Update views.py

TharukaN99 committed:
Update views.py

TharukaN99 committed:
Update views.py

TharukaN99 committed:
Update views.py

TharukaN99 committed:
Update views.py

TharukaN99 committed:
Update edit_profile.dart for conflict

TharukaN99 committed:
Update views.py reslove conflict

TharukaN99 committed:
Update views.py reslove conflict

TharukaN99 committed:
Update views.py reslove conflict

TharukaN99 committed:
Update views.py reslove conflict

TharukaN99 committed:
Update views.py reslove conflict

TharukaN99 committed:
Update views.py reslove conflict

TharukaN99 committed:
Update views.py reslove conflict

TharukaN99 committed:
Update views.py reslove conflict

TharukaN99 committed:
Update views.py reslove conflict

TharukaN99 committed:
Update views.py reslove conflict

TharukaN99 committed:
Merge branch 'main' into main

Thumbness committed:
Merge branch 'main' into main

Thumbness reviewed (approved):
Thanks for the cleanup
